src/NServiceBus.Core/Pipeline/PipelineModelBuilder.cs

```diff
@@ -22,9 +23,23 @@ public List<RegisterStep> Build()
             var listOfBeforeAndAfterIds = new List<string>();
 
             // Let's do some validation too
+            if (addOrReplaceSteps.Any(addOrReplaceStep => addOrReplaceStep.RegisterStep.StepId != addOrReplaceStep.ReplaceStep.ReplaceId))
```

Comment:
Looking at the code in AddOrReplaceStep.cs can this really happen?

Reply:
Not anymore, thanks for pointing that out. It could happen but I refactored the AddOrReplaceStep creation